wsmaster/che-core-api-devfile/src/test/java/org/eclipse/che/api/devfile/server/validator/DevfileSchemaValidatorTest.java

```diff
@@ -66,60 +66,60 @@ public void shouldThrowExceptionOnValidationNonValidDevfile(
+        "Devfile schema validation failed. Error: /devfile object instance has properties which are not allowed by the schema: \\[\"unknown\"\\]$"
       },
       // Tool model testing
       {
         "tool/devfile_missing_tool_name.yaml",
-        "Devfile schema validation failed. Errors: \\[object has missing required properties \\(\\[\"name\"\\]\\)\\]$"
+        "Devfile schema validation failed. Error: /devfile/tools/0 object has missing required properties \\(\\[\"name\"\\]\\)$"
       },
       {
         "tool/devfile_missing_tool_type.yaml",
-        "Devfile schema validation failed. Errors: \\[object has missing required properties \\(\\[\"type\"\\]\\)\\]$"
+        "Devfile schema validation failed. Error: /devfile/tools/0 object has missing required properties \\(\\[\"type\"\\]\\)$"
       },
       {
         "tool/devfile_tool_with_undeclared_field.yaml",
-        "Devfile schema validation failed. Errors: \\[object instance has properties which are not allowed by the schema: \\[\"unknown\"\\]\\]$"
+        "Devfile schema validation failed. Error: /devfile/tools/0 object instance has properties which are not allowed by the schema: \\[\"unknown\"\\]$"
       },
       // Command model testing
       {
         "command/devfile_missing_command_name.yaml",
-        "Devfile schema validation failed. Errors: \\[object has missing required properties \\(\\[\"name\"\\]\\)\\]$"
+        "Devfile schema validation failed. Error: /devfile/commands/0 object has missing required properties \\(\\[\"name\"\\]\\)$"
       },
       {
         "command/devfile_missing_command_actions.yaml",
-        "Devfile schema validation failed. Errors: \\[object has missing required properties \\(\\[\"actions\"\\]\\)\\]$"
+        "Devfile schema validation failed. Error: /devfile/commands/0 object has missing required properties \\(\\[\"actions\"\\]\\)$"
       },
       {
         "command/devfile_multiple_commands_actions.yaml",
-        "Devfile schema validation failed. Errors: \\[array is too long: must have at most 1 elements but instance has 2 elements\\]$"
+        "Devfile schema validation failed. Error: /devfile/commands/0/actions array is too long: must have at most 1 elements but instance has 2 elements$"
       },
       // cheEditor/chePlugin tool model testing
       {
         "editor_plugin_tool/devfile_editor_tool_with_missing_id.yaml",
-        "Devfile schema validation failed\\. Errors: \\[instance failed to match exactly one schema \\(matched 0 out of 3\\)\\]"
+        "Devfile schema validation failed\\. Errors: \\[instance failed to match exactly one schema \\(matched 0 out of 3\\),/devfile/tools/0 object has missing required properties \\(\\[\"id\"\\]\\),/devfile/tools/0 object has missing required properties \\(\\[\"local\"\\]\\),/devfile/tools/0 object has missing required properties \\(\\[\"image\",\"memoryLimit\"\\]\\)\\]"
```

Comment:
It definitely looks much detailed than before but It still does not look clear =(
I think `if`/`else` statement is the only thing that can help us to provide pretty good UX